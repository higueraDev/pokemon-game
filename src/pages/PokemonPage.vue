<script>
import PokemonPicture from "@/components/PokemonPicture.vue";
import PokemonOptions from "@/components/PokemonOptions.vue";
import getPokemonOptions from "@/helpers/getPokemonOptions";

export default {
	components: { PokemonPicture, PokemonOptions },
	data() {
		return {
			pokemonArr: [],
			pokemon: null,
			showPokemon: false,
			showAnswer: false,
			message: "",
		};
	},
	mounted() {
		this.mixPokemonArr();
	},
	methods: {
		async mixPokemonArr() {
			this.pokemonArr = await getPokemonOptions();
			const randomInteger = Math.floor(Math.random() * 4);
			this.pokemon = this.pokemonArr[randomInteger];
		},
		checkAnswer(id) {
			this.showPokemon = true;
			if (id == this.pokemon.id) {
				this.message = `Correcto, es ${this.pokemon.name}`;
			} else {
				this.message = `Oops!, era ${this.pokemon.name}...`;
			}
			this.showAnswer = true;
		},
		newGame() {
			this.mixPokemonArr();
			this.pokemon = null;
			this.showPokemon = false;
			this.showAnswer = false;
		},
	},
};
</script>

<template>
	<h1 v-if="!pokemon">Espere por favor...</h1>
	<div v-else>
		<h1>¿Quien es este pokemón?</h1>
		<PokemonPicture
			:showPokemon="showPokemon"
			:pokemonId="pokemon.id"
		></PokemonPicture>
		<PokemonOptions
			@selection="checkAnswer"
			:pokemons="pokemonArr"
		></PokemonOptions>
		<template v-if="showAnswer" class="fade-in">
			<h2>
				{{ message }}
			</h2>
			<button @click="newGame">Nuevo Juego</button>
		</template>
	</div>
</template>
<style></style>
