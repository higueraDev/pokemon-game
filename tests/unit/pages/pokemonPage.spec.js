import PokemonPage from "@/pages/PokemonPage";
import { mount, shallowMount } from "@vue/test-utils";
import { pokemons } from "../mocks/pokemons.mock";

describe("PokemonPage Page", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallowMount(PokemonPage);
	});
	test("should match snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});
	test("should call mixPokemonArray", () => {
		const mixPokemonArraySpy = jest.spyOn(
			PokemonPage.methods,
			"mixPokemonArr"
		);
		shallowMount(PokemonPage);
		expect(mixPokemonArraySpy).toHaveBeenCalled();
	});
	test("should match snapshot when data loads", () => {
		//mount() Monta los componentes con componentes hijos
		const wrapper = mount(PokemonPage, {
			data() {
				return {
					pokemonArr: pokemons,
					pokemon: pokemons[0],
					showPokemon: false,
					showAnswer: false,
					message: "",
				};
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});
});
