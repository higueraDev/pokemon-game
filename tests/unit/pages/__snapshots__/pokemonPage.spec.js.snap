// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Page should match snapshot 1`] = `<h1>Espere por favor...</h1>`;

exports[`PokemonPage Page should match snapshot when data loads 1`] = `
<div>
  <h1>¿Quien es este pokemón?</h1>
  <div class="pokemon-container" data-testid="pokemonPicture"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemón"></div>
  <div class="options-container" data-testid="pokemonOptions">
    <ul>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
